libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -666,17 +666,13 @@ bool NavEKF2::InitialiseFilter(void)
         // check if there is enough memory to create the EKF cores
         if (hal.util->available_memory() < sizeof(NavEKF2_core)*num_cores + 4096) {
             initFailure = InitFailures::NO_MEM;
-            gcs().send_text(MAV_SEVERITY_CRITICAL, "NavEKF2: not enough memory available");
```

Comment:
I'm not sure we should change behaviour to keep attempting to allocate memory for the EKF.  If it's failed once it's never going to recover.

Maybe it would be better to create a new boolean, perhaps called "bool cores_allocated" (put it right below the definition for core in AP_navEKF2.h?) and then in the InitialiseFilter, change the "if (core == nullptr)" check so that it also checks this new boolean.  and then we always set the boolean to true if we've ever attempted that allocation part of the code.